daemon/container.go

```diff
@@ -222,6 +222,11 @@ func (daemon *Daemon) setHostConfig(container *container.Container, hostConfig *
 	container.Lock()
 	defer container.Unlock()
 
+	if daemon.configStore.Init && hostConfig.Init == nil {
```

Comment:
This doesn't feel right; can we not do this where `dockerd --init` sets things today?

Reply:
Do you means  add this code for docker-cli or put it somewhere else in dockerd? If add it for docker-cli,docker-cli will need to read daemon.json directly or indirectly

Reply:
I add it in getInspectData func at docker/docker/daemon/inspect.go, it will not hardcode into  hostconfig.json.  this can solve my problem, but I think it is not perfect.

Reply:
use  docker run --init ,container always run with docker-init,even if restart this container.
if use init:true in daemon.json, container will run with  docker-init, otherwise  container will not run with  docker-init.
It's performance just depend on daemon.json.